openmc/mgxs_library.py

```diff
@@ -2099,6 +2429,107 @@ def get_by_name(self, name):
                 result = xsdata
         return result
 
+    def convert_representation(self, target_representation, num_polar=None,
+                               num_azimuthal=None):
+        """Produce a new MGXSLibrary object with the same data, but converted
+        to the new representation
+
+        Parameters
+        ----------
+        target_representation : {'isotropic', 'angle'}
+            Representation of the MGXS (isotropic or angle-dependent flux
+            weighting).
+        num_polar : int, optional
+            Number of equal width angular bins that the polar angular
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+        num_azimuthal : int, optional
+            Number of equal width angular bins that the azimuthal angular
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+
+        Returns
+        -------
+        openmc.MGXSLibrary
+            Multi-group Library with the same data as self, but represented as
+            specified in :param:`target_representation`.
+
+        """
+
+        check_value('target_representation', target_representation,
```

Comment:
Can lines 2459-2465 be removed since the`XSdata.convert_representation(...)` method also performs the same error checking?